aggregate_root/spec/aggregate_root_spec.rb

```diff
@@ -229,8 +229,8 @@
       order.apply(Orders::Events::OrderExpired.new)
       expect(order.status).to eq(:expired)
 
-      expect(order.private_methods).to include(:"on_Orders::Events::OrderCreated")
-      expect(order.private_methods).to include(:"on_Orders::Events::OrderExpired")
```

Comment:
potential collision when same class name but different namespace ie `Orders::Created`, `Customers::Created` (not very domain-specific example but just to illustrate the problem).

Reply:
Possible. Should that happen in aggregate? Should `on` behave differently than `non-on` in the same default strategy?